src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -901,7 +908,8 @@ export class TerminalService implements ITerminalService {
 			this._terminalShellTypeContextKey,
 			this._terminalAltBufferActiveContextKey,
 			this._configHelper,
```

Comment:
considered adding this to shellLaunchConfig, but figured that wouldn't add much

Reply:
shell launch config generally shouldn't be modified once the process gets created as it's purpose is to create/recreate the process, not for checking its result